### PR TITLE
Fix default expiry for timestamp-based RFD 153-style resources

### PR DESCRIPTION
**Pull request by Tener**
RFD 153-style resources are using `timestamppb.Timestamp` instead of `time.Time` for expiry. The zero value for this type is 01/01/1970. This is a problem for resources without explicit expiry set: they become instantly obsolete on creation. To avoid this problem, we will now check for nil expiry explicitly, and default it to `time.Time{}`.

### CODE REVIEW

**Comment by Tener** [comment]
Split from https://github.com/gravitational/teleport/pull/37803.

**Review by rosstimothy — APPROVED** [review]

File: api/types/resource.go [hunk 1]
```diff
@@ -747,7 +747,14 @@ func GetExpiry(v any) time.Time {
 	case Resource:
 		return r.Expiry()
 	case ResourceMetadata:
-		return r.GetMetadata().Expires.AsTime()
+		// ResourceMetadata uses *timestamppb.Timestamp instead of time.Time. The zero value for this type is 01/01/1970.
+		// This is a problem for resources without explicit expiry set: they'd become obsolete on creation.
+		// For this reason, we check for nil expiry explicitly, and default it to time.Time{}.
```
Comment (codingllama) [inline api/types/resource.go:752]: Mention this in the function godoc?

**Review by codingllama — APPROVED** [review]
Thanks for the fix and for the split!

File: api/types/resource.go (hunk 1 above)
Comment (Tener) [reply api/types/resource.go:752]: The godoc update got lost in the split, I'll fix it.

**Pull request merged**